### PR TITLE
Add authz-td-migration user guide test.

### PR DESCRIPTION
**Pull request by justinpettit**

### CODE REVIEW

File: tests/security/authz_td_migration/scripts/authz_td_migration.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+set -e
+set -u
+set -o pipefail
+
+source "${REPO_ROOT}/content/en/docs/tasks/security/authorization/authz-td-migration/snips.sh"
+
+max_attempts=10
+
+snip_before_you_begin_3
```
Comment (ericvn) [inline tests/security/authz_td_migration/scripts/authz_td_migration.sh:27]: Do we need to do snip_before_you_begin_2?

Comment (justinpettit) [reply tests/security/authz_td_migration/scripts/authz_td_migration.sh:27]: snip_before_you_begin_2 sets up the namespace which needs to be a blocking call, so it's called as part of the Go program itself.  This task is really slow to execute, so I think it's timing out.  I have a patch that will improve debugging, which I plan to send out shortly, and then I'll confirm that.

File: tests/security/authz_td_migration/main_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authztdmigration
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
```
Comment (frankbu) [inline tests/security/authz_td_migration/main_test.go:21]:
```suggestion
```

File: tests/security/authz_td_migration/scripts/authz_td_migration.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+
+set -e
+set -u
+set -o pipefail
+
+source "${REPO_ROOT}/content/en/docs/tasks/security/authorization/authz-td-migration/snips.sh"
+source "${REPO_ROOT}/tests/util/samples.sh"
+
+max_attempts=10
```
Comment (frankbu) [inline tests/security/authz_td_migration/scripts/authz_td_migration.sh:26]: I'm removing the `max_attempts` argument from the `_run_and_verify` functions in #7440. You can override the default 5 attempts by setting the `VERIFY_RETRIES` environment variable.

Comment (justinpettit) [reply tests/security/authz_td_migration/scripts/authz_td_migration.sh:26]: Sounds good.  This test is particularly slow, so it needs a higher retry count.  Hopefully we can start rolling out tests with the new istioctl "wait" mechanism to improve this situation.

**Comment by frankbu** [comment]
@justinpettit This needs to be updated to use the latest new-and-improved framework style (https://github.com/istio/istio.io/tree/master/tests#test-authoring-overview). If you aren't able to do it soon, please close this PR and I can take what you've done, convert to the new style, and finish it for you. Thanks.

**Comment by frankbu** [comment]
Test added in #7653

Note that the last check (trustDomainAliases) in this test isn't working, so it's currently commented out.

**Pull request closed without merging**